### ISSUE TITLE
[Feature request (or question)]Shortcuts

### ISSUE THREAD

**Issue by mbastias** [starik222/BooruDatasetTagManager#16]
I was using this on my last LoRA and it works very good. But I wanted to know if there's shortcuts for things like Apply changes, Add or Delete tag when one is editing a picture. If not, I would like to ask for the request, because one is typing the tags and then has to switch to the mouse to Apply or keep adding tags. If they were somewhere and I skipped them, I apologize.

**Comment by neojam** [comment]
Yes having shortcuts for most common operations would be nice to have.

Also a checkbox under "Settings>General" to "Always save changes" would be nice. 
Currently the popup-message "The list of Tags have been changed. Save Changes?" is quite annoying, because in 99.9% of cases i always want to save the made changes.

**Comment by mbastias** [comment]
Alright, insert and delete, only missing the apply shortcut then, but the not asking to save as a workaround is fine too. Thanks a lot for the tool, is a breeze to use and very light against the extension.

**Comment by aphaits** [comment]
Shortcuts would help so much.

**Comment by jojje** [comment]
I added a shortcut functionality in [a fork](https://github.com/jojje/BooruDatasetTagManager/tree/master) since I was missing that myself.

In particular I needed to be able to navigate the images, tags, swap between the two panes, and add, edit and apply tags without having to use a mouse. So those were the shortcuts I added.

If there are elements in that [feature branch](https://github.com/jojje/BooruDatasetTagManager/tree/feat/keyboard-shortcuts) the author wants to use, go ahead. 

For the starik; I don't think pulling the feature in as-is would be appropriate, since it was implemented as orthogonally as possible to your code base, in order to 1) minimize the risk of interference with your existing key-event subscriptions, and 2) make it easier for me to follow upstream changes in the branch (avoid merge conflicts). 

If you're looking to use bits of this, I'd recommend just re-using the command architecture, but integrating it _properly_ in the upstream code. I.e. factoring out functionality that currently resides in the forms themselves, into separate actions (commands), and using windows forms event subscription instead of the low level win32 API trampoline I was forced to use to not break your event listeners.

Also, I noticed there is a [feature](https://github.com/starik222/BooruDatasetTagManager/issues/55) request for Undo. I had that type of functionality in mind when designing the shortcut feature, and is why the shortcuts are based on the command pattern. It would be trivial to implement undo of most of your functionality was moved into separate commands. You'd just add a reverse operation for each command to their respective command classes and put the executed commands in a stack (or list). Undo would be just popping each command and executing its reverse operation.

I.e. right now I only implemented the _forward_ operation (`execute`). To support undo, a reverse operation like `revert` or `undo` ( :) ) would be paired with the forward operation. CTRL-Z would just pop the last command from the stack and execute `revert`.
